ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/ethernet_interfaces.py

```diff
@@ -88,14 +88,27 @@ def _get_ethernet_interface_cfg(self, adapter: dict, node_index: int, connected_
         channel_group_id = get(adapter, "port_channel.channel_id", default=default_channel_group_id)
         short_esi = self._get_short_esi(adapter, channel_group_id)
 
+        # check lengths of lists
+        nodes_length = len(adapter["switches"])
+        if len(adapter["switch_ports"]) != nodes_length or (
+            "descriptions" in adapter and "description" not in adapter and len(adapter["descriptions"]) != nodes_length
+        ):
```

Comment:
Why check for `description` not in adapter here? if it is present but the `descriptions` list is of the right length then `descriptions` will be used and if it is present but `descriptions` is not of the correct length then it will raise. So `description` in adapter is useless here.

Reply:
done